pkg/didcomm/packer/legacy/authcrypt/unpack.go

```diff
@@ -66,68 +68,70 @@ func getCEK(recipients []recipient, km kms.KeyManager) (*[chacha.KeySize]byte, e
 
 	recKeyIdx, err := km.FindVerKey(candidateKeys)
 	if err != nil {
-		return nil, fmt.Errorf("no key accessible %w", err)
+		return nil, nil, fmt.Errorf("no key accessible %w", err)
 	}
 
 	recip := recipients[recKeyIdx]
 	recKey := recip.Header.KID
 
 	recCurvePub, err := km.ConvertToEncryptionKey(base58.Decode(recKey))
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
-	sender, err := decodeSender(recip.Header.Sender, recCurvePub, km)
+	senderPub, senderPubCurve, err := decodeSender(recip.Header.Sender, recCurvePub, km)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	nonceSlice, err := base64.URLEncoding.DecodeString(recip.Header.IV)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	encCEK, err := base64.URLEncoding.DecodeString(recip.EncryptedKey)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	b, err := kms.NewCryptoBox(km)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
-	cekSlice, err := b.EasyOpen(encCEK, nonceSlice, sender, recCurvePub)
+	cekSlice, err := b.EasyOpen(encCEK, nonceSlice, senderPubCurve, recCurvePub)
 	if err != nil {
-		return nil, fmt.Errorf("failed to decrypt CEK: %s", err)
+		return nil, nil, fmt.Errorf("failed to decrypt CEK: %s", err)
 	}
 
 	var cek [chacha.KeySize]byte
 
 	copy(cek[:], cekSlice)
 
-	return &cek, nil
+	return &cek, senderPub, nil
 }
 
-func decodeSender(b64Sender string, pk []byte, km kms.KeyManager) ([]byte, error) {
+func decodeSender(b64Sender string, pk []byte, km kms.KeyManager) ([]byte, []byte, error) {
 	encSender, err := base64.URLEncoding.DecodeString(b64Sender)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	b, err := kms.NewCryptoBox(km)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
-	senderSlice, err := b.SealOpen(encSender, pk)
+	senderPub, err := b.SealOpen(encSender, pk)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
-	senderData := base58.Decode(string(senderSlice))
+	senderData := base58.Decode(string(senderPub))
+
+	senderPubCurve, err := cryptoutil.PublicEd25519toCurve25519(senderData)
 
-	return cryptoutil.PublicEd25519toCurve25519(senderData)
+	return senderPub, senderPubCurve, err
```

Comment:
`senderPubCurve` is raw bytes of the public encryption key
while `senderPub` seems to be the byte representation of the `base58` encoding of the public signing (verification) key

I guess the tests are passing as senderPub is sent to the caller of Unpack which expects a base58 encoded key... either make this clear in the comment, or return senderData (raw bytes of public signing key) and do base58.Encode after  getting the output of Unpack().

but returning keys should be of the same type (either return both base58 encoded or both raw bytes, but not a mix of 1 base58 encoded and the other not.. we prefer returning both raw bytes).

in short, return `senderData` instead of `senderPub` and call base58.Encode() at https://github.com/hyperledger/aries-framework-go/pull/880/files#diff-68bdfb240e9da84a7ae7cdd37c85a8e4R154 

and make sure the `jwe` Packer implementation returns the same raw byte (not base58 encoded) to keep the same behaviour as in `legacy` implementation

Reply:
[Updated](https://github.com/hyperledger/aries-framework-go/compare/b34c5cd85fb496c143d5c08f96451194c349a7c8..c11501105b1b19784aa4a95fe77f516372579ac5#diff-b6253114caf4247940f4442d6515375aR134).